crates/aptos/src/move_tool/mod.rs

```diff
@@ -985,7 +1051,35 @@ impl FromStr for FunctionArgType {
             "u128" => Ok(FunctionArgType::U128),
             "hex_array" => Ok(FunctionArgType::HexArray),
             "raw" => Ok(FunctionArgType::Raw),
-            str => Err(CliError::CommandArgumentError(format!("Invalid arg type '{}'.  Must be one of: ['address','bool','hex','hex_array','string','u8','u64','u128','raw']", str))),
+            str => {
+                // If it's a vector, go one level inside
+                if str.starts_with("vector<") && str.ends_with('>') {
+                    let arg = FunctionArgType::from_str(&str[7..str.len() - 1])?;
+
+                    // String gets confusing on parsing by commas
+                    if arg == FunctionArgType::String {
+                        return Err(CliError::CommandArgumentError(
+                            "vector<string> is not supported".to_string(),
+                        ));
+                    } else if arg == FunctionArgType::Raw {
+                        return Err(CliError::CommandArgumentError(
+                            "vector<raw> is not supported".to_string(),
+                        ));
+                    } else if matches!(arg, FunctionArgType::Vector(_)) {
+                        return Err(CliError::CommandArgumentError(
+                            "nested vector<vector<_>> is not supported".to_string(),
+                        ));
+                    } else if arg == FunctionArgType::HexArray {
+                        return Err(CliError::CommandArgumentError(
+                            "nested vector<hex_array> is not supported".to_string(),
+                        ));
+                    }
+
+                    Ok(FunctionArgType::Vector(Box::new(arg)))
+                } else {
+                    Err(CliError::CommandArgumentError(format!("Invalid arg type '{}'.  Must be one of: ['address','bool','hex','hex_array','string','u8','u64','u128','raw', 'vector<inner_type>']", str)))
```

Comment:
This feels prone it getting out of date. Strum or something would be overkill but can we add a comment in FunctionTypeArg that says to go update this if they add / remove / modify a variant there?